### ISSUE TITLE
WARNING - The model and loaded state dict do not match exactly, with a lot of 'unexpected key'

### ISSUE THREAD

**Issue by DrX2020** [open-mmlab/mmdetection#4945]
Case 1: 
Previously I trained a Faster R-CNN detector on my own dataset by setting the default 'pretrained='torchvision://resnet50'' and 'load_from=None' (I was not fine-tuning) and only got two 'unexpected key': fc.weight, fc.bias. It seemed normal because I had different number of classes.

Case 2:
I also trained a Cascade R-CNN (HRNet backbone) model by setting 'pretrained='open-mmlab://msra/hrnetv2_w40'' and 'load_from=None' and got a warning with a pile of 'unexpected key': incre_modules.0.0.conv1.weight, incre_modules.0.0.bn1.weight, etc. It managed to do the whole traing and gave a result, though.

Case 3:
I also tried to fine-tune a Cascade R-CNN (HRNet backbone) model I trained before by setting 'pretrained=None' and 'load_from=path of the trained Cascade R-CNN (HRNet backbone) model' and got a warning with even more 'unexpected key', two different piles, actually. Again, it managed to do the whole training and gave a result.

To see if the parameters are successfully loaded, I set 'pretrained=None' and 'load_from=None' in case 2 and case 3 and the loss was nan from the start. In other word, the training cannot be done. 

So, despite of the warnings, case 2 and case 3 did load something because they were different from training with 'pretrained=None' and 'load_from=None'. How much should I worry about the warnings?

**Comment by ZwwWayne** [comment]
This is fine as the pre-trained backbone contains unnecessary keys like FC. Some keys in the head will also mismatch during fine-tuning because the num_classes may be different.
